contracts/actions/mcd/McdPayback.sol

```diff
@@ -14,31 +14,55 @@ import "./helpers/McdHelper.sol";
 contract McdPayback is ActionBase, McdHelper {
     using TokenUtils for address;
 
+    /// @param _vaultId Id of the vault
+    /// @param _amount Amount of dai to be payed back
+    /// @param _from Where the Dai is pulled from
+    /// @param _mcdManager The manager address we are using
+    struct Params {
+        uint256 vaultId;
+        uint256 amount;
+        address from;
+        address mcdManager;
+    }
+
     /// @inheritdoc ActionBase
     function executeAction(
         bytes[] memory _callData,
         bytes[] memory _subData,
         uint8[] memory _paramMapping,
         bytes32[] memory _returnValues
     ) public payable override returns (bytes32) {
-        (uint256 vaultId, uint256 amount, address from, address mcdManager) =
-            parseInputs(_callData);
+        Params memory inputData = parseInputs(_callData);
 
-        vaultId = _parseParamUint(vaultId, _paramMapping[0], _subData, _returnValues);
-        amount = _parseParamUint(amount, _paramMapping[1], _subData, _returnValues);
-        from = _parseParamAddr(from, _paramMapping[2], _subData, _returnValues);
+        inputData.vaultId = _parseParamUint(
+            inputData.vaultId,
+            _paramMapping[0],
+            _subData,
+            _returnValues
+        );
+        inputData.amount = _parseParamUint(
+            inputData.amount,
+            _paramMapping[1],
+            _subData,
+            _returnValues
+        );
+        inputData.from = _parseParamAddr(
+            inputData.from,
+            _paramMapping[2],
+            _subData,
+            _returnValues
+        );
 
-        amount = _mcdPayback(vaultId, amount, from, mcdManager);
+        _mcdPayback(inputData);
 
-        return bytes32(amount);
+        return bytes32(inputData.amount);
```

Comment:
functions pass structs as reference, so the inputData.amount should be the amount changed on line 84